tests/run/staticmethod_pure.py

```diff
@@ -0,0 +1,30 @@
+import sys
+# behaviour is significantly different between Python2 and 3
+if sys.version_info[0] == 2:
+    __doc__ = u"""
+>>> class1.plus1(1) #doctest: +IGNORE_EXCEPTION_DETAIL
+Traceback (most recent call last):
+    ...
+TypeError: unbound
+>>> class2.plus1(1) #doctest: +IGNORE_EXCEPTION_DETAIL
+Traceback (most recent call last):
+    ...
+TypeError: unbound
+"""
+else:
+    __doc__ = u"""
+>>> class1.plus1(1)
+2
+>>> class2.plus1(1)
+2
+"""
+
+
+def f_plus(a):
+    return a + 1
+
+class class1:
+    plus1 = f_plus
+
+class class2(object):
+    plus1 = f_plus
```

Comment:
These examples do not use static methods at all, the use entirely normal methods. I can see that this test was previously in `staticmethod.pyx`, but moving it out calls for a better name of this test file now. Maybe `method_assignment.py` ?

Reply:
OK - I've made that change. The other option might be to put them in with the function_as_method_T494 py and pyx file which cover a similar (but not quite identical) theme. It'd also provide somewhere to put `class3`.

Reply:
I've moved them there. While doing that I've done a bit of tidying up of the class names, and added a test for the old `binding(False)` behaviour (which I assume will continue to exist if specifically requested)